src/components/ProductAddForm.js

```diff
@@ -0,0 +1,49 @@
+import React, { Component } from 'react'
+import { validateFields } from '../utils/validator'
+import { db } from '../firebase'
+
+const byPropKey = (propertyName, value) => () => ({
+  [propertyName]: value,
+})
+
+const INITIAL_STATE = {
+  name: '',
+  error: null,
+}
+
+export default class ProductAddForm extends Component {
+  state = { ...INITIAL_STATE }
+
+  onSubmit = event => {
+    const { name } = this.state
+    db.create('Product', {
+      name,
+    })
+    this.setState({ ...INITIAL_STATE })
+    event.preventDefault()
+  }
+
+  render() {
+    const { name, error } = this.state
+
+    const isInvalid = !validateFields({
```

Comment:
You should validate on input change or submit and hold validation result in component's state.